### ISSUE TITLE
Want to help get this working again

### ISSUE THREAD

**Issue by struktured** [projectM-visualizer/frontend-qt#12]
Are the issues in this repo reflective of what has to be fixed to get back to a functional app?  Would like get this working again so please know how I can help!

**Comment by revmischa** [comment]
I expect some changes are needed to support the new 4.x API
Welcome to come [chat with us on Discord](https://discord.gg/N9DyQfCH4j)
I'm working on https://github.com/projectM-visualizer/frontend-sdl-rust with SDL3

**Comment by kblaschke** [comment]
More or less, I'd say.

The main things requiring work would be:
- Updating the projectM integration to use the new 4.0 API, as Mischa said. That's not hard to do. May either use the new playlist library, or just manage preset switching in the Qt app itself, as it already has its own playlist functionality.
- Making audio capturing more portable, e.g. by using a wrapper library such as PortAudio.
- Updating the whole Qt code to work with the most recent Qt 6 versions, including making proper use of QOpenGlWidget.

No one I know of is currently working on the codebase, so feel free to take it over. We'll answer questions and review PRs!

**Comment by struktured** [comment]
I would argue that #1 is the most important since it gets back to functional again if that's completed.

**Comment by kblaschke** [comment]
Absolutely. Third one is also important, as the application exposed weird rendering issues last time I've worked on the Qt 6 upgrades. These might now be mitigated by rewriting the rendering process in libprojectM. From what I could see in RenderDoc it was an issue that some (not all) VBOs/VAOs initialized at preset load time were already lost when rendering a frame, but they weren't actively deleted by projectM.

**Comment by struktured** [comment]
Ok it's functional again! Can we update the about section to indicate the project is no longer broken? @kblaschke 

I'm also open to feature requests at this point.

**Comment by kblaschke** [comment]
There are a few other issues in this repo, if you want, you can browse through, close outdated ones and fix a few if possible.

I guess for feature requests, it might help to create an issue for this and pin it in the issue tracker.

If you're missing issue triage permissions here, drop me a message, I'll have a look then.
